src/battle_script_commands.c

```diff
@@ -6168,7 +6172,12 @@ static void Cmd_recordlastability(void)
 {
     gActiveBattler = GetBattlerForBattleScript(gBattlescriptCurrInstr[1]);
     RecordAbilityBattle(gActiveBattler, gLastUsedAbility);
+
+#ifndef UBFIX
```

Comment:
This seems like a bug, not undefined behavior.  Luckily, it's never used in any battle scripts.